.github/workflows/test_project.yaml

```
@@ -0,0 +1,42 @@
+name: CI
+
+on: [push]
```

```suggestion
on:
  push:
    branches: [main]
  pull_request:
```

This will currently not run for PRs from my fork.

Otherwise, if you prefer to run CI on your pushes, it think it's better to just keep the duplicated workflows (`on: [push, pull_request]`).

---

Ok, then let's run it only on pull request and avoid direct pushes to main